servicetalk-transport-api/src/main/java/io/servicetalk/transport/api/TransportObservers.java

```diff
@@ -36,18 +36,27 @@ public static TransportObserver asSafeObserver(final TransportObserver observer)
         if (observer instanceof CatchAllTransportObserver) {
             return observer;
         }
+        if (observer instanceof BiTransportObserver) {
+            // BiTransportObserver is always safe
+            return observer;
+        }
         return new CatchAllTransportObserver(observer);
     }
 
     /**
-     * Combines two {@link TransportObserver}s into a single {@link TransportObserver}.
+     * Combines multiple {@link TransportObserver}s into a single {@link TransportObserver}.
      *
      * @param first the {@link TransportObserver} that will receive events first
      * @param second the {@link TransportObserver} that will receive events second
-     * @return a {@link TransportObserver} that delegates all invocations to the {@code first} and {@code second}
-     * {@link TransportObserver}s
+     * @param others additional {@link TransportObserver}s that will receive events
+     * @return a {@link TransportObserver} that delegates all invocations to the provided {@link TransportObserver}s
      */
-    public static TransportObserver biTransportObserver(final TransportObserver first, final TransportObserver second) {
-        return new BiTransportObserver(first, second);
+    public static TransportObserver combine(final TransportObserver first, final TransportObserver second,
```

Comment:
Why not just have `combine(final TransportObserver... observers)` ?

Also, may be modify the `BiTransportObserver` to accept an array instead of two observers.

Reply:
At least two observer are required to combine. Good to highlight that in the API instead of verifying at runtime.

Having an array inside `BiTransportObserver` will require an array inside every observer level, producing boilerplate at every level to traverse the array. Let's start with simplified approach and we can reconsider if necessary.